data/registry_model/registry_proxy_model.py

```diff
@@ -385,6 +385,9 @@ def _update_manifest_for_tag(
 
         placeholder = manifest.internal_manifest_bytes.as_unicode() == ""
         if up_to_date and not placeholder:
+            if tag.expired:
+                upstream_manifest = self._pull_upstream_manifest(repo_ref.name, manifest_ref)
```

Comment:
darn I said that wrong, I meant to check that`upstream_manifest` _was_ none before pulling to avoid pulling it twice, something like:
```
if upstream_manifest is None:
    upstream_manifest = self._pull_upstream_manifest(repo_ref.name, manifest_ref)
```
sorry about that

Reply:
makes sense, thanks for pointing it out!